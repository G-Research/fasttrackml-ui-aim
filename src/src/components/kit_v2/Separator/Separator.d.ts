import { SeparatorProps } from '@radix-ui/react-separator';

import { CSS } from 'config/stitches/types';

export interface ISeparatorProps extends SeparatorProps {
  /**
   * @description The color of the separator
   * @example 'primary'
   * @default 'gray'
   */
  color?: CSS['color'];
  /**
   * @description The margin of the separator
   * @example '10px'
   * @default '0'
   * @type {CSS['margin']}
   */
  margin?: CSS['margin'];
  /**
   * @description The css prop of the separator
   * @example { margin: '10px' }
   * @default {}
   * @type {CSS}
   */
  css?: CSS;
}
