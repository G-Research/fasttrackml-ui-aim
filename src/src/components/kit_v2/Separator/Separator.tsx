import React from 'react';

import { SeparatorRoot } from './Separator.style';
import { ISeparatorProps } from './Separator.d';

/**
 * Separator component
 * @param {string} color - color of the separator line
 * @param {string} margin - margin of the separator line
 * @param {CSS} css - css of the separator line
 * @param {ISeparatorProps} props - HTML attributes
 * @returns {React.FunctionComponentElement<React.ReactNode>}
 * @example
 * <Separator />
 * <Separator color="$secondary30" margin="10px" />
 * <Separator color="$secondary30" margin="10px" css={{ backgroundColor: 'red' }} />
 * <Separator color="$secondary30" margin="10px" css={{ backgroundColor: 'red' }} orientation="vertical" />
 */
function Separator({
  orientation = 'horizontal',
  color = '$secondary30',
  margin,
  css,
  ...props
}: ISeparatorProps): React.FunctionComponentElement<React.ReactNode> {
  return (
    <SeparatorRoot
      {...props}
      className='Separator'
      css={{
        backgroundColor: color,
        margin: orientation === 'horizontal' ? `${margin} 0` : `0 ${margin}`,
        ...css,
      }}
      orientation={orientation}
      data-testid='separator'
    />
  );
}

Separator.displayName = 'Separator';
export default React.memo(Separator);
