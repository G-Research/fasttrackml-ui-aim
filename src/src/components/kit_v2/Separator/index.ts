import Separator from './Separator';

export * from './Separator.d';
export default Separator;
