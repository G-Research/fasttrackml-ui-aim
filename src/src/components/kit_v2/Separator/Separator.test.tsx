import { render } from '@testing-library/react';

import Separator from './Separator';

// Test of Separator component

describe('<Separator />', () => {
  test('Renders Correctly', () => {
    const { asFragment } = render(<Separator />);
    expect(asFragment()).toMatchSnapshot();
  });

  // test color prop
  test('color prop works properly', () => {
    const { getByTestId } = render(<Separator color='$primary100' />);
    const separator = getByTestId('separator');
    expect(separator).toHaveStyle('background-color: $primary100');
  });

  // test orientation prop
  test('orientation prop works properly', () => {
    const { getByTestId } = render(<Separator orientation='vertical' />);
    const separator = getByTestId('separator');
    expect(separator).toHaveAttribute('data-orientation', 'vertical');
  });

  // test margin prop
  test('margin prop works properly', () => {
    const { getByTestId } = render(<Separator margin='$4' />);
    const separator = getByTestId('separator');
    expect(separator).toHaveStyle('margin: "0 $4');
  });

  // test css prop
  test('css prop works properly', () => {
    const { getByTestId } = render(<Separator css={{ color: 'red' }} />);
    const separator = getByTestId('separator');
    expect(separator).toHaveStyle("color: 'red'");
  });
});
