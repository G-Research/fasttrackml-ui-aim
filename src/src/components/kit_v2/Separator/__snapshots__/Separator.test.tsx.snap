// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Separator /> Renders Correctly 1`] = `
<DocumentFragment>
  <div
    class="aim-ui-c-gzAwqY aim-ui-c-gzAwqY-ifXICFP-css Separator"
    data-orientation="horizontal"
    data-testid="separator"
    role="separator"
  />
</DocumentFragment>
`;
